src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -76,12 +86,57 @@ public int Execute()
 
             WriteReport(migrationReport);
 
+            MoveProjectJsonArtifactsToBackup(_doBackup, migrationReport);
+
             return migrationReport.FailedProjectsCount;
         }
 
-        private void WriteReport(MigrationReport migrationReport)
+        private void MoveProjectJsonArtifactsToBackup(bool doBackup, MigrationReport migrationReport)
         {
+            if (!doBackup)
+            {
+                return;
+            }
+            
+            if (migrationReport.FailedProjectsCount > 0)
+            {
+                return;
+            }
+            
+            _backupDirectory.Create();
+
+            var globalJson = Path.Combine(_workspaceDirectory.FullName, GlobalSettings.FileName);
+
+            if (File.Exists(globalJson))
+            {
+                File.Move(globalJson, Path.Combine(_backupDirectory.FullName, GlobalSettings.FileName));
+            }
+            
+            foreach (var report in migrationReport.ProjectMigrationReports)
+            {
+                var projectDirectory = PathUtility.EnsureTrailingSlash(report.ProjectDirectory);
```

Comment:
And then, the meat of it into a BackupProjectFiles, that would hold the code inside this loop.